src/frontend/screens/Settings/ProjectSettings/DeviceName/index.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from 'react';
+import {Alert, ScrollView, StyleSheet} from 'react-native';
+import {NativeStackNavigationOptions} from '@react-navigation/native-stack';
+import {MessageDescriptor, defineMessages, useIntl} from 'react-intl';
+import {useForm} from 'react-hook-form';
+
+import {
+  NativeNavigationComponent,
+  NativeRootNavigationProps,
+} from '../../../../sharedTypes';
+import {
+  useDeviceInfo,
+  useEditDeviceInfo,
+} from '../../../../hooks/server/deviceInfo';
+import {Text} from '../../../../sharedComponents/Text';
+import {HookFormTextInput} from '../../../../sharedComponents/HookFormTextInput';
+import {IconButton} from '../../../../sharedComponents/IconButton';
+import {EditIcon, SaveIcon} from '../../../../sharedComponents/icons';
+import {FieldRow} from './FieldRow';
+import {BLACK} from '../../../../lib/styles';
+
+const m = defineMessages({
+  title: {
+    id: 'screens.Setting.ProjectSettings.DeviceName.title',
+    defaultMessage: 'Device Name',
+  },
+  deviceNameDisplayHeader: {
+    id: 'screens.Setting.ProjectSettings.DeviceName.deviceNameDisplayHeader',
+    defaultMessage: 'Your Device Name',
+  },
+  deviceNameEditHeader: {
+    id: 'screens.Setting.ProjectSettings.DeviceName.deviceNameEditHeader',
+    defaultMessage: 'Edit Device Name',
+  },
+  discardChangesAlertTitle: {
+    id: 'screens.Setting.ProjectSettings.DeviceName.discardChangesAlertTitle',
+    defaultMessage: 'Discard changes?',
+  },
+  discardChangesAlertConfirmText: {
+    id: 'screens.Setting.ProjectSettings.DeviceName.discardChangesAlertConfirmText',
+    defaultMessage: 'Discard Changes',
+  },
+  discardChangesAlertCancelText: {
+    id: 'screens.Setting.ProjectSettings.DeviceName.discardChangesAlertCancelText',
+    defaultMessage: 'Continue Editing',
+  },
+});
+
+export type NavigationParams = {mode: 'display' | 'edit'};
+
+export const DeviceName: NativeNavigationComponent<'DeviceName'> = ({
+  route,
+  navigation,
+}) => {
+  const {mode} = route.params;
+  const {data} = useDeviceInfo();
+
+  return (
+    <ScrollView contentContainerStyle={styles.container}>
+      {data?.name &&
```

Comment:
not really sure what the best approach to this is i.e. when data is potentially loading or fails to load. In theory, i think we don't even render the app if that's the case, but just thinking more generally...

Reply:
In this case, would it not make sense to show a blank name? Device Name is optional in schema (to allow for the instance where you freshly install a CoMapeo and and you are instantly added to a project, but have not created your name yet) 

In otherwords it is not necessarily a loading error. It could be the case that someone just does not have a name.